### ISSUE TITLE
Allow the user to add custom themes

### ISSUE THREAD

**Issue by K4LCIFER** [LemmyNet/lemmy-ui#2162]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

Currently, the following themes are available:

![image](https://github.com/LemmyNet/lemmy-ui/assets/7829231/22260265-6de8-448a-907f-6832c826402f)

It would be a nice feature to allow the user to add their own custom themes.

This is possibly related to #694.

**Comment by travis-jeans** [comment]
I think this is different to issue 694 (my understanding of the other thread is that it's about letting instance admins select a default light and dark mode rather than one single theme). 

At the moment if Lemmy users on an instance want to add their own theme to the drop down list, they can't unless they get an Admin to upload it first. 

If there was a way for users to upload a theme themself with "Load Theme From File" button or something like that, it could load a custom theme (css file) which would only be available for that user and stored in user preferences or something, maybe it could help with theme development and also help deal with the issues of users loading custom stylesheets over the top of default themes which can cause style conflicts.

**Comment by dessalines** [comment]
Yes we can't allow users to upload themes, these need to be included in the lemmy-ui server.
